lib/vector-buffers/src/variants/disk_v2/writer.rs

```diff
@@ -582,6 +582,7 @@ where
     /// Errors can occur during the encoding, serialization, or I/O stage.  If an error occurs
     /// during any of these stages, an appropriate error variant will be returned describing the error.
     #[instrument(skip(self, record), level = "trace")]
+    #[cfg(test)]
```

Comment:
This is only used in tests?

Reply:
Apparently. @tobz to confirm this makes sense.

Reply:
That's correct.

This is the `RecordWriter`, which is a fancy little wrapper over the buffered writer, and handles encoding and serialization and all of that. It exposes two methods -- `archive_record` and `flush_record` -- which actually get used by `Writer`, and `RecordWriter::write_record` only gets used in tests just to reduce the amount of code.

This change is fine for now, but I'll make a note that using `archive_record`/`flush_record` directly in the tests would also be sufficient and that this method can be removed entirely.